components/UserProfile.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect } from "react";
+import {
+  View,
+  Text,
+  Image,
+  ScrollView,
+  TouchableOpacity,
+  SafeAreaView,
+} from "react-native";
+import { SHADOWS } from "../theme";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { baseUrl } from "../env";
+import { useNavigation } from "@react-navigation/native";
+import { dummyData } from "../DummyData";
+import axios, { AxiosError } from "axios";
+
+const users = dummyData;
+
+type User = {
+  id: string;
+  username: string;
+  nickname: string;
+  avatarUrl: number;
+  posts: number[];
+  isFollowing: boolean;
+};
+
+type RootStackParamList = {
+  FollowerList: { type: number; users: any };
+  Authentification: undefined;
+  EditProfile: { user: any };
+};
+
+type Props = {
+  personal: boolean;
+  user: User;
+};
+type NavigationType = StackNavigationProp<RootStackParamList, "FollowerList">;
+
+const UserProfile: React.FC<Props> = ({ user, personal }) => {
+  const navigation = useNavigation<NavigationType>();
+  const [userNotFound, setUserNotFound] = useState(false);
+  const [notAuthorized, setNotAuthorized] = useState(false);
+  const [username, setUsername] = useState("");
+  const [nickname, setNickname] = useState("");
+  const [status, setStatus] = useState("");
+  // const [profilePicture, setProfilePicture] = useState("");
+  const [followerCount, setFollowerCount] = useState("");
+  const [followingCount, setFollowingCount] = useState("");
+  const [postsCount, setPostsCount] = useState("");
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      try {
+        const response = await axios.get(`${baseUrl}users/:username`);
```

Comment:
Ich würde vorschlagen, dass wir fetch von React Native direkt nutzen, da es einfach leichtgewichtiger ist und wir die extra funktionen von Axios sowieso nicht verwenden.

Lässt sich aber in der 3. Iteration auch noch Refactoren, also passt alles für mich